azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -246,27 +238,50 @@ private static Props setAzkabanProps(final Path jobtypePluginPath) {
     final String[] args = {CONF_ARG, CONF_DIR};
     final Props props = AzkabanServer.loadProps(args);
 
-    // Set db stuff.
     return new Props(props, propsMap);
   }
 
-  // Submit flow
-  public void submitFlow(final int execId) throws ExecutorManagerException {
-    final FlowRunner flowRunner = createFlowRunner(execId);
-    submitFlowRunner(flowRunner);
+  private static void setInjector(final Props azkabanProps){
+    // Inject AzkabanCommonModule
+    final Injector injector = Guice.createInjector(
+            new AzkabanCommonModule(azkabanProps)
+    );
+    SERVICE_PROVIDER.setInjector(injector);
+  }
+
+  /**
+   * Submit flow
+   * Creates and submits the FlowRunner.
+   * @param execId Execution Id of the flow.
+   * @throws ExecutorManagerException
+   */
+  private void submitFlow(final int execId) throws ExecutorManagerException {
+    submitFlowRunner(createFlowRunner(execId));
   }
 
-  // create Flow Runner
+  /**
+   * create Flow Runner
+   * @param execId Execution Id of the flow.
+   * @return FlowRunner object.
+   * @throws ExecutorManagerException
+   */
   private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerException {
     final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(execId);
     if (flow == null) {
-      throw new ExecutorManagerException("Error loading flow with exec " + execId);
+      logger.error("Error loading flow with execution Id " + execId);
+      throw new ExecutorManagerException("Error loading flow for exec: " + execId +
+              ". Terminating flow container launch");
     }
 
     // Update the status of the flow from DISPATCHING to PREPARING
     flow.setStatus(Status.PREPARING);
```

Comment:
Bring this to main method. As soon as you get valid execution id, you should set status to PREPARING by capturing flow object over there. createFlowRunner should accept ExecutableFlow flow in parameter.